app/components/SecureTokenBox.tsx

```diff
@@ -0,0 +1,140 @@
+import { Input } from "@nextui-org/input";
+import { Code } from "@nextui-org/code";
+import { Button } from "@nextui-org/button";
+import { Textarea } from "@nextui-org/input";
+import { useState, useCallback, useMemo, ChangeEvent } from "react";
+
+import {Card, Link, Select, SelectItem} from "@nextui-org/react";
+import { AggregatorInputParams} from "../utils/types";
+import { generateSecureToken } from "../utils/queries";
+import { BLOCKCHAIN_LIST } from "../utils/blockchains";
+
+export default function SecureTokenBox({ aggregatorInputs, showBuyQuoteURLText, blockchains }: { aggregatorInputs?: AggregatorInputParams, showBuyQuoteURLText?: boolean, blockchains?: string[]}) {
+  const [secureToken, setSecureToken] = useState("");
+  const [ethAddress, setEthAddress] = useState("");
+
+  const [blockchainOption, setBlockchainOption] = useState("");
+
+  const setBlockchain = useCallback((event: ChangeEvent<HTMLSelectElement>) => {
+    setBlockchainOption(event.target.value);
+  }, []);
+
+  const secureTokenWrapper = useCallback(async () => {
+    const response = await generateSecureToken({ethAddress, blockchains: showBuyQuoteURLText ? blockchains : [blockchainOption.toLowerCase()]})
+    console.log("generateSecureToken");
+    try {
+      if (response) {setSecureToken(response);} else {setSecureToken('')}
+    } catch (error) {
+      console.log(error)
+      alert(error);
+      console.error(error);
+    }}, [ethAddress,  showBuyQuoteURLText, blockchains, blockchainOption]);
+
+  const linkReady = useMemo(() => secureToken.length > 0, [secureToken]);
+
+  const link = useMemo(() => {
+    if (!linkReady) return "Generate a secure token first to create your one time URL";
+    return (
+      "https://pay.coinbase.com/buy/select-asset?sessionToken=" + secureToken + 
```

Comment:
Consider using [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) to construct the query string, then `https://pay.coinbase.com/buy/select-asset?${searchParams}`